site/gdocs/OwidGdocPage.tsx

```diff
@@ -76,6 +77,81 @@ function getPageDesc(gdoc: OwidGdocUnionType): string | undefined {
         .exhaustive()
 }
 
+interface JsonLdAuthor {
+    "@type": "Person" | "Organization"
+    name: string
+    url?: string
+}
+
+function makeJsonLdAuthors(
+    baseUrl: string,
+    gdoc: OwidGdocPostInterface
+): JsonLdAuthor[] {
+    return gdoc.content.authors.map((gdocAuthor) => {
+        if (gdocAuthor.toLowerCase().includes("our world in data")) {
+            return {
+                "@type": "Organization",
+                name: "Our World in Data",
+                url: baseUrl,
+            }
+        }
+        const author: JsonLdAuthor = {
+            "@type": "Person",
+            name: gdocAuthor,
+        }
+        const linkedAuthor = gdoc.linkedAuthors?.find(
+            (linkedAuthor) => linkedAuthor.name === gdocAuthor
+        )
+        // URLs serve as unique IDs for authors, so we don't use the team page
+        // URL for authors, who don't have their own page.
+        if (linkedAuthor?.slug) {
+            author.url = getCanonicalUrl(baseUrl, {
+                slug: linkedAuthor.slug,
+                content: { type: OwidGdocType.Author },
+            })
+        }
+        return author
+    })
+}
+
+function JsonLdArticle({
+    gdoc,
+    baseUrl,
+    imageUrl,
+}: {
+    gdoc: OwidGdocPostInterface
+    baseUrl: string
+    imageUrl?: string
+}) {
+    const data = {
+        "@context": "https://schema.org",
+        "@type": "Article",
+        headline: gdoc.content.title,
+        image: imageUrl ? [imageUrl] : [],
+        datePublished: gdoc.publishedAt,
+        dateModified: gdoc.updatedAt,
+        author: makeJsonLdAuthors(baseUrl, gdoc),
+    }
+    return (
+        <script
+            type="application/ld+json"
+            dangerouslySetInnerHTML={{
+                __html: JSON.stringify(data),
+            }}
+        />
+    )
+}
+
+function isPostPredicate(
```

Comment:
You might want to colocate and differentiate this from [checkIsGdocPostExcludingFragments](https://github.com/owid/owid-grapher/blob/master/packages/%40ourworldindata/utils/src/Util.ts#L1881)

A shame that we need 3 of these now, but oh well! 🙃

Reply:
Yeah... well the function param is a different type, and it's not clear we'll need the predicate anywhere else, so I'd keep it here for now.